MapboxNavigation/RouteVoiceController.swift

```diff
@@ -110,6 +110,7 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
     deinit {
         suspendNotifications()
         speechSynth.stopSpeaking(at: .immediate)
+        audioPlayer?.delegate = nil
```

Comment:
Shouldn’t AVAudioPlayer be responsible for nilling out its own delegate when it deinits? `AVAudioPlayer.delegate` is unsafe-unowned rather than weak, which is unfortunate, but it’d be surprising if AVAudioPlayer doesn’t nil it out anyways in its deinit method. Is there something else keeping the audio player alive, perhaps?

Reply:
Yeah let's go with #1572 first.